### ISSUE TITLE
[BUG] Checking backslash (`'\\'`) in an if check causes erroneous cppfront compiler errors

### ISSUE THREAD

**Issue by JakeShirley** [hsutter/cppfront#892]
**Describe the bug**
Checking a backslash ('\') character value in an `if` check causes cppfront to think there are brace mismatches.

**To Reproduce**
Example code:
```cpp2
main: () -> int = {
    testValue : char = '\\';

    if testValue == '\\' {
        return 1;
    }

    return 0;
}
```

Expected Behavior:
Code compiles and runs, or alternatively gives an accurate error of what and why the error occurred.

Observed Results:
cppfront fails with the following error:
```
main.cpp2:9:0: error: closing } does not match a prior {
main.cpp2: error: unexpected end of source file
```

**Additional context**
A friend and I are using cppfront in Advent of Code this year and both (separately) hit this on the 2023 Day 16 problem  😅 .

**Comment by JohelEGP** [comment]
It seems like you're using an outdated version.
It works on Compiler Explorer: <https://cpp2.godbolt.org/z/fsvzKefsr>.

**Comment by JakeShirley** [comment]
Welp, you're right!

[Updating to the latest ](https://github.com/hsutter/cppfront/compare/5ea5d5c3d145ffc288326113d848040d9d77f75e...faf57aff350bbfbd615711021ded0cd225d1c6e0)seems to make the error go away.

Thanks for the quick reply, closing.

**Comment by JakeShirley** [comment]
To close the loop, this was fixed in c1d3e0edc8b2e2a2c035f68b6c0012c5b7049a07
